src/engine/index/manager.rs

```
@@ -0,0 +1,689 @@
+        key: String,
+        row_path: String,
+    ) -> errors::Result<()> {
+        // Update memory first
+        {
+            let mut indices = self.indices.write().await;
+            let tree = indices
+                .get_mut(index_name)
+                .ok_or_else(|| ExecuteError::wrap(format!("index '{}' not found", index_name)))?;
+
+            tree.insert(key.clone(), row_path.clone())
+                .map_err(|e| ExecuteError::wrap(e))?;
```

Clippy (1.96.0)

[warning] redundant closure

(warning)